src/core/Slider/Slider.tsx

```diff
@@ -0,0 +1,494 @@
+  }
+  const distances = values.map((value) => Math.abs(value - pointerValue));
+  const smallest = Math.min(...distances);
+  return distances.indexOf(smallest);
+};
+
+const getDefaultTrackDisplay = (
+  trackDisplayMode: TrackDisplayMode,
+  values: number[],
+) => {
+  if ('auto' !== trackDisplayMode) {
+    return trackDisplayMode;
+  }
+
+  return values.length % 2 ? 'even-segments' : 'odd-segments';
+};
+
+const roundValueToClosestStep = (value: number, step: number, min: number) => {
+  return Math.round((value - min) / step) * step + min;
+};
+
+const formatNumberValue = (
+  value: number,
+  step: number,
+  numDecimals: number,
+) => {
+  if (Number.isInteger(step)) {
+    return value.toFixed(0);
+  }
+  return value.toFixed(numDecimals);
+};
+
+/**
+ * Focus specified thumb on Slider control
+ * @param sliderContainer containing div hold slider
+ * @param activeIndex index of the thumb
+ */
+export const focusThumb = (
+  sliderContainer: HTMLDivElement,
+  activeIndex: number,
+) => {
+  const doc = sliderContainer.ownerDocument;
+  if (
+    !sliderContainer.contains(doc.activeElement) ||
+    Number(doc.activeElement?.getAttribute('data-index')) !== activeIndex
+  ) {
+    const thumbToFocus = sliderContainer.querySelector(
+      `[data-index="${activeIndex}"]`,
+    );
+    thumbToFocus && (thumbToFocus as HTMLElement).focus();
+  }
+};
+
+/**
+ * Properties for Slider component
+ */
+export type SliderProps = {
+  /**
+   * Set focus on first thumb in slider element.
+   * @default false
+   */
+  setFocus?: boolean;
+  /**
+   * Minimum slider value.
+   * @default 0
+   */
+  min?: number;
+  /**
+   * Maximum slider value.
+   * @default 100
+   */
+  max?: number;
+  /**
+   * Array of one or more values to show.
+   */
+  values: number[];
+  /**
+   *  Determines which segments are shown with color.
+   * 'none' - no colored tracks are displayed.
+   * 'auto' - segment display is based on number of values.
+   * 'odd-segments'- colored tracks shown in segments 1,3,5, etc.
+   *                 Default if number of thumbs values are even.
+   * 'even-segments'- colored tracks shown in segments 0,2,4, etc.
+   *                 Default if number of thumbs values are odd.
+   * @default 'auto'
+   */
+  trackDisplayMode?: TrackDisplayMode;
+  /**
+   * Step increment controls what values are allowed and the amount the value will
+   * change when left and right arrows are pressed when a Thumb has focus.
+   * @default 1
+   */
+  step?: number;
+  /**
+   * Forces control to be displayed in a disabled state where no interactive value
+   * changes are allowed.
+   *  @default false
+   */
+  disabled?: boolean;
+  /**
+   * Props to override default for showing a tooltip when Thumb is active or has focus.
+   */
+  tooltipProps?: Partial<Omit<TooltipProps, 'content' | 'children'>>;
+  /**
+   * Function that can generate tooltip content.
+   */
+  tooltipRenderer?: (val: number, step: number) => React.ReactNode;
```

Comment:
If you change tooltipProps approach to use `(index: number) => TooltipProps`, it would make it similar to `thumbProps`, and you won't need `tooltipRenderer` anymore.

Reply:
I think it is a good idea to get props by index, but I do not want to make the generation of the tooltip text anymore complicated then it is today.  So this mean I would want to pass the function (index: number, val: number, step: number) so tooltipProps function.